sqlglot/dialects/athena.py

```diff
@@ -36,7 +37,29 @@ class Generator(Trino.Generator):
             exp.FileFormatProperty: lambda self, e: f"'FORMAT'={self.sql(e, 'this')}",
         }
 
+        def generate(self, expression: exp.Expression, copy: bool = True) -> str:
```

Comment:
why didn't you use standard transform in TRANSFORMS ?

Reply:
...standard transform? I didn't see one that could intercept the entire expression before any of the other transforms.

The quoting to use is determined by the root node, so attempting to figure it out at the `exp.Identifier` level seemed like more work than simply figuring it out at the start and storing it as metadata